numpy/lib/_function_base_impl.py

```diff
@@ -5418,43 +5418,49 @@ def insert(arr, obj, values, axis=None):
     -----
     Note that for higher dimensional inserts ``obj=0`` behaves very different
     from ``obj=[0]`` just like ``arr[:,0,:] = values`` is different from
-    ``arr[:,[0],:] = values``.
+    ``arr[:,[0],:] = values``. This is because of the difference between basic
+    and advanced :ref:`indexing <basics.indexing>`.
```

Comment:
I think this appropriately points the user to a reference if they aren't sure why things behave differently. It's not a one line fix, but the example added below hopefully addresses that.